tests/platform_tests/mellanox/test_thermal_control.py

```diff
@@ -89,11 +81,31 @@ def test_set_psu_fan_speed(duthosts, rand_one_dut_hostname, mocker_factory):
 
     logging.info('Mock FAN presence...')
     single_fan_mocker.mock_presence()
-    assert wait_until(THERMAL_CONTROL_TEST_WAIT_TIME, THERMAL_CONTROL_TEST_CHECK_INTERVAL, check_cooling_cur_state, duthost, 10, operator.ne), \
-        'Current cooling state is {}'.format(get_cooling_cur_state(duthost))
-    logging.info('Wait {} seconds for the policy to take effect...'.format(THERMAL_CONTROL_TEST_CHECK_INTERVAL))
-    time.sleep(THERMAL_CONTROL_TEST_CHECK_INTERVAL)
+    wait_until(THERMAL_CONTROL_TEST_WAIT_TIME, THERMAL_CONTROL_TEST_CHECK_INTERVAL, check_cooling_cur_state, duthost, 10, operator.ne)
+    logging.info('Wait {} seconds for the policy to take effect...'.format(THERMAL_CONTROL_TEST_WAIT_TIME * 2))
+    # We have to wait THERMAL_CONTROL_TEST_WAIT_TIME * 2 seconds long here because:
```

Comment:
rephrase comments. 
why change for all systems if there is a system which behaves differently?
please suggest alternative how to get the valid numbers.

Reply:
Please see new comments.